docs/.vuepress/containers/examples/examples.js

```diff
@@ -0,0 +1,136 @@
+      block: true,
+      hidden: false
+    }
+  ];
+};
+
+const getPreviewTab = (id, cssContent, htmlContent, code) => {
+  return {
+    type: 'html_block',
+    tag: '',
+    attrs: null,
+    map: [],
+    nesting: 0,
+    level: 1,
+    children: null,
+    content: `
+      <tab name="Preview" id="preview-tab-${id}">
+        <style v-pre>${cssContent}</style>
+        <div v-pre>${htmlContent}</div>
+        <ScriptLoader v-if="$parent.$parent.isScriptLoaderActivated('${id}')" code="${code}"></ScriptLoader>
+      </tab>
+    `,
+    markup: '',
+    info: '',
+    meta: null,
+    block: true,
+    hidden: false
+  };
+};
+
+module.exports = {
+  type: 'example',
+  render(tokens, index, opts, env) {
+    const token = tokens[index];
+    const m = token.info.trim().match(EXAMPLE_REGEX);
+    const version = env.relativePath.split('/')[0];
+
+    if (token.nesting === 1 && m) { // open preview
+      let [, , id, klass, preset, args] = m;
+
+      id = id ? id.substring(1) : 'example1';
+      klass = klass ? klass.substring(1) : '';
+      preset = preset ? preset.substring(1) : 'hot';
+      args = args || '';
+
+      const htmlPos = args.match(/--html (\d*)/)?.[1];
+      const htmlIndex = htmlPos ? index + Number.parseInt(htmlPos, 10) : 0;
+      const htmlToken = htmlPos ? tokens[htmlIndex] : undefined;
+      const htmlContent = htmlToken
+        ? htmlToken.content
+        : `<div id="${id}" class="hot ${klass}"></div>`;
+
+      const cssPos = args.match(/--css (\d*)/)?.[1];
+      const cssIndex = cssPos ? index + Number.parseInt(cssPos, 10) : 0;
+      const cssToken = cssPos ? tokens[cssIndex] : undefined;
+      const cssContent = cssToken ? cssToken.content : '';
+
+      const jsPos = args.match(/--js (\d*)/)?.[1] || 1;
+      const jsIndex = index + Number.parseInt(jsPos, 10);
+      const jsToken = tokens[jsIndex];
+      const jsContent = jsToken.content;
+
+      const activeTab = args.match(/--tab (code|html|css|preview)/)?.[1] || 'code';
+      const noEdit = !!args.match(/--no-edit/)?.[0];
+
+      const code = buildCode(id + (preset.includes('angular') ? '.ts' : '.jsx'), jsContent, env.relativePath);
+      const encodedCode = encodeURI(`useHandsontable('${version}', function(){${code}}, '${preset}')`);
+
+      [htmlIndex, jsIndex, cssIndex].filter(x => !!x).sort().reverse().forEach((x) => {
+        tokens.splice(x, 1);
+      });
+
+      const newTokens = [
+        ...tab('Code', jsToken),
+        ...tab('HTML', htmlToken),
+        ...tab('CSS', cssToken),
+        getPreviewTab(id, cssContent, htmlContent, encodedCode)
+      ];
+
+      tokens.splice(index + 1, 0, ...newTokens);
+
+      return `
+          ${!noEdit ? jsfiddle(id, htmlContent, jsContent, cssContent, version, preset) : ''}
+          <tabs
+            :options="{ useUrlFragment: false, defaultTabHash: '${activeTab}' }"
+            cache-lifetime="0"
+            @changed="$parent.$parent.codePreviewTabChanged(...arguments, '${id}')"
```

Comment:
I fully agree with you! That is an anty pattern.

 However, in this specific case, it is required and acceptable. Because this code is generated and added into the whole app during the build. We (me and @budnix ) want to keep logic in the common Vue component (Page.vue) which isn't dynamically generated. That makes it much easier to maintain.

Reply:
Additionally, `examples.js` is hard to maintain by its use case.

It is run during build (in markdown loader). Then it builds code examples, wraps them into `script loader` component. The output(as string) goes into VueLoader, then it is built again and returns SSR files (HTML). And then, when served, when the end-user goes into the specific page: it unwraps code examples, loads their dependencies, and runs them, manages their lifecycle, and descruting if the page is changed.

Using `$parent` follows the complexity of the whole that process.

Reply:
cc @budnix

Reply:
It doesn't hold back the initial release. Let's do it at your earliest convenience @Siemienik

Reply:
Reported https://github.com/handsontable/handsontable/issues/8429.